ISISScientificComputing/autoreduce
# Add interactive plots in web app AR-1376

### Summary of work
Adds support for visualising interactive plots. This requires that they are plotted using `plotly` and saved into a file via plotly's figure method: `fig.to_json()`. 

Static plot support is untouched.

### How to test your work
These instructions build on https://github.com/ISISScientificComputing/autoreduce/wiki/Just-Use-VSCode-&-Docker

Start containers:
- ActiveMQ - `Run ActiveMQ Docker`

Migrate the DB + fixtures with task `Migrate database (with PR fixtures)`

Run the debug "Webapp Docker attach" task. This will mount the folder at `repo-root/reduced-data` to `/instrument` in the container, so you can just put this sample JSON inside the `reduced-data` folder:

[reduced-data.zip](https://github.com/ISISScientificComputing/autoreduce/files/6440764/reduced-data.zip)

The exact location it needs to be can be seen in the [fixture](https://github.com/ISISScientificComputing/autoreduce/pull/1328/files#diff-4c09c19744a78af717bc61d2c00c80251dcbd09c85464aa8c92999cd7ea6252dR33), but note that the `/instrument` is inside the container.

Once the webapp is running, open it and go to http://127.0.0.1:8000/runs/TestInstrument/99999/0/, you should see this:

![image](https://user-images.githubusercontent.com/9135965/117434786-02c44500-af25-11eb-9e5c-3ba1f437a16c.png)

To do:
- [x] Update selenium plot tests 
- [x] Make & update 21.1.1 release notes
